### ISSUE TITLE
MD054: Markdown alerts are seen as URLs in shortcut format

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#483]
### rumdl version

0.1.36

### Installation method

mise

### Bug description

MD054 (https://github.com/rvben/rumdl/blob/fb2532f1ae6d2b6e329a2225c6517705ffecd72d/docs/md054.md) allows projects to establish a consistent link style.

However, the rule can catch instances of alert syntax that looks like like `shortcut` link syntax (https://github.com/rvben/rumdl/blob/fb2532f1ae6d2b6e329a2225c6517705ffecd72d/docs/md054.md#link-and-image-styles).

Examples where this alert syntax extends basic CommonMark:

- https://docs.gitlab.com/user/markdown/#alerts
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
- https://gohugo.io/render-hooks/blockquotes/#basic-syntax

Because this syntax is seen across different flavours of Markdown, it might be worth this syntax being supported by default.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: heading
   ---
   
   > [!note]
   > Content for the note.
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [MD054]
   shortcut = false
   ```
1. Run `rumdl check` on the Markdown file.

### Expected behavior

Given the scenario above, `rumdl` should return no error.

### Actual behavior

Given the scenario above, `rumdl` returns this error:

```shell
test.md:5:3: [MD054] Link/image style 'shortcut' is not allowed
```

### Configuration

```toml
[MD054]
shortcut = false
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in [v0.1.37](https://github.com/rvben/rumdl/releases/tag/v0.1.37). MD054 now skips alert/callout syntax like `[!NOTE]` in shortcut link detection. Thanks for reporting @eread!
